packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -27,32 +27,46 @@ const GQL_TYPESCRIPT_DATA_SCHEMA_TYPE_MAP = {
  * @returns Typescript data schema property in TS Node format
  */
 const createProperty = (field: Field): ts.Node => {
-  const typeExpression = createDataType(field.type);
+  const typeExpression = createDataType(field);
   return ts.factory.createPropertyAssignment(ts.factory.createIdentifier(field.name), typeExpression as ts.Expression);
 };
 
 /**
  * Creates a typescript data schema type from internal SQL schema representation
  * Example typescript data schema type output: `a.string().required()`
- * @param type SQL IR field type
+ * @param field SQL IR field
  * @returns Typescript data schema type in TS Node format
  */
-const createDataType = (type: FieldType): ts.Node => {
-  if (type.kind === 'Scalar') {
+const createDataType = (field: Field): ts.Node => {
+  const sequenceRegex = /^nextval\(.+::regclass\)$/;
```

Comment:
[Regex Explain](https://regexr.com/871ku)

I went back and forth on a few approaches to this check but I eventually felt that a quick REGEX is the simplest surefire way to identify `SERIAL` fields. In the future I think this logic could be extracted into something like `if (canLetDBGenerate(field))` which evaluates rules `isSerial(field) || isUserDefinedFunction(field)` etc... But I felt that might premature for this implementation.

Reply:
For more context, here's what a value for `column_default` looks like on a serial field when we query the DB schema:
`nextval('serial_table_number_seq'::regclass)`.

Reply:
Can we find documentation that this is the canonical way to determine whether a field is `serial`? In the past we've relied on experimentally-verified "rules" that turned out to be incomplete, so I'd love to get ahead of that this time.

Reply:
Added main reply with my research/argument later in thread. I don't believe there is a truly canonical way to identify serial but we can identify if a column is backed by a sequence.